### ISSUE TITLE
Smart punctuation does not get applied in nested inlines (e.g. emphasis)

### ISSUE THREAD

**Issue by neongreen** [mmark-md/mmark-ext#10]
```
$ mmark --ext-punctuation
*Hello...*
<p><em>Hello...</em></p>

$ mmark --ext-punctuation
Hello...
<p>Hello…</p>
```

**Comment by neongreen** [comment]
This probably applies to some other extensions that use `inlineTrans` and `blockTrans`. Also, this behavior was surprising to me and I think it should be mentioned in the docs for `inline/blockTrans`.

**Comment by neongreen** [comment]
I fixed a similar problem in my extensions' code by writing `inlineTrans $ Uniplate.transform $ \case -> ...`, but I'm not sure what your preferred solution would be.

**Comment by mrkkrp** [comment]
This is a bug in `mmark`. Similar to what I fixed in version 0.0.5.2 ([see here](https://github.com/mmark-md/mmark/blob/master/CHANGELOG.md#mmark-0052)). I'll get to this.

**Comment by mrkkrp** [comment]
Closing in favor of mmark-md/mmark#47.

**Comment by mrkkrp** [comment]
Fixed in `mmark-0.0.5.6`.
